sklearn/compose/_column_transformer.py

```diff
@@ -409,15 +420,28 @@ def fit_transform(self, X, y=None):
         result = self._fit_transform(X, y, _fit_transform_one)
 
         if not result:
+            self._update_fitted_transformers([])
             # All transformers are None
             return np.zeros((X.shape[0], 0))
 
         Xs, transformers = zip(*result)
 
+        # determine if concatenated output will be sparse or not
+        if all(sparse.issparse(X) for X in Xs):
+            self.sparse_output_ = True
+        elif any(sparse.issparse(X) for X in Xs):
+            nnz = sum(X.nnz if sparse.issparse(X) else X.size for X in Xs)
+            total = sum(X.shape[0] * X.shape[1] if sparse.issparse(X)
+                        else X.size for X in Xs)
+            density = nnz / total
+            self.sparse_output_ = density < self.sparse_threshold
```

Comment:
"if the density is at most this value" implies `<=` here. Perhaps the wording can be improved. A test should check this case to avoid regression.

Reply:
I adapted the test such that the density of the data was a 'round' float so I can test it. For now I updated the wording to match the code, but I can of course also change the code to be `<=`.